toolboxes/mri_core/mri_core_utility.cpp

```diff
@@ -6,8 +6,246 @@
 
 #include "mri_core_utility.h"
 #include "hoNDArray_elemwise.h"
+#include "mri_core_kspace_filter.h"
+#include <ctime>
 
 namespace Gadgetron
 {
+    template <typename T>
+    void detect_readout_sampling_status(const hoNDArray<T>& data, hoNDArray<float>& sampled)
```

Comment:
This function should return sampled, rather than taking it as input. It also seems like the type of sampled should perhaps be hoNDArray<bool>?